src/components/upload/video-uploader.tsx

```diff
@@ -55,13 +50,7 @@ const VideoUploader = ({
       onError={(message) => console.log(message)}
       onFinish={(signResult, file) => {
         const fileUrl = signResult.signedUrl.split('?')[0]
-        setLessonMetadata((prevState: Array<object>) => [
-          ...prevState,
-          {
-            title: file.name,
-            fileMetadata: {fileName: file.name, signedUrl: fileUrl},
-          },
-        ])
+        dispatch({type: 'finalize', file, fileUrl})
```

Comment:
`setLessonMetadata` is replaced with a new action `finalize` that can be dispatched, reducing the props surface area of this component.